docs(microservices) add microservices-timeout page (#1258)
nestjs/docs.nestjs.com: content/microservices/basics.md

```

The `data` property is the message payload sent by the message producer. The `pattern` property is the pattern used to identify an appropriate handler to handle the incoming message.

### Handling timeouts

> Sometimes, microservices are down or not available. You can handle these cases using the `RxJS` timeout operator in the call. If the microservice does not respond to the request within a certain time, an Exception is thrown, which can be caught and handled appropriately.

Reviewer: ```suggestion
In distributed systems, sometimes microservices might be down or not available. To avoid infinitely long waiting, you can use Timeouts. A timeout is an incredibly useful pattern when communicating with other services. To apply timeouts to your microservice calls, you can use the `RxJS` timeout operator. If the microservice does not respond to the request within a certain time, an exception is thrown, which can be caught and handled appropriately.
```

> To solve this problem you have to use [rxjs](https://github.com/ReactiveX/rxjs) package. Just use the `timeout` operator in the pipe:

Reviewer 2: Can we get consistent use of `RxJS` on the page? Also, can we say `you have to use the RxJS package`?

```

The `data` property is the message payload sent by the message producer. The `pattern` property is the pattern used to identify an appropriate handler to handle the incoming message.

> ### Timeout

Reviewer: ```suggestion
### Handling timeouts
```
what do you think^?

Sometimes, microservices are down or not available. You can handle these cases using the `RxJS` timeout operator in the call. If the microservice does not respond to the request within a certain time, an Exception is thrown, which can be caught and handled appropriately.

To solve this problem you have to install [rxjs](https://github.com/ReactiveX/rxjs) package:

```bash
npm install rxjs

> ```

Reviewer: `rxjs` is a `peerDependency` of NestJS packages. It should be installed already

> Then just add a call of timeout method to the pipe:

Reviewer 3: ```suggestion
Then just use the `timeout` operator in the pipe:
```

```typescript
@@filename()
import { timeout } from 'rxjs/operators';
// ...
this.client
      .send<TResult, TInput>(pattern, data)
      .pipe(timeout(5000))
      .toPromise();
@@switch
import { timeout } from 'rxjs/operators';
// ...
this.client
      .send(pattern, data)
      .pipe(timeout(5000))
      .toPromise();
```

> After 5 seconds, if microservice isn't responding, it will throw an error:
> ```bash
> [Nest] 17544   - 05/24/2020, 8:08:27 AM   [ExceptionsHandler] Timeout has occurred +5009ms
> ```

Reviewer: ```suggestion
After 5 seconds, if the microservice isn't responding, it will throw an error.
```
Perhaps, we can avoid showing the error log - it will vary by logger used by the application.
